File: api/app/user/infrastructure/adapter/inbound/api/message/BaseUserResponse.py
from typing import Self

from pydantic import BaseModel

from app.user.domain.entity import User
from app.user.domain.enum import Role


class BaseUserResponse(BaseModel):
    id: str
    username: str
    email: str
    role: Role

    @classmethod
    def from_domain_entity(cls, user: User) -> Self:
        return cls(
            id=str(user.id),
            username=user.username,
            email=str(user.email),
            role=user.role,
        )

File: console/src/data/repositories/messages/common/base-user-response.ts
import type { Role, User } from '@/domain/models/user'

export interface BaseUserResponse {
  id: string
  username: string
  email: string
  role: Role
  createdAt: string
  lastSignIn: string | null
}

export function toUser(response: BaseUserResponse): User {
  return {
    id: response.id,
    username: response.username,
    email: response.email,
    role: response.role,
    createdAt: new Date(response.createdAt),
    lastSignIn: response.lastSignIn ? new Date(response.lastSignIn) : null,
  }
}
